src/modules.ts

```diff
@@ -109,32 +109,63 @@ export async function generateInjectManifest(options: ResolvedVitePWAOptions, vi
 
   const { format, plugins, rollupOptions } = options.injectManifestRollupOptions
 
-  await build({
-    root: viteOptions.root,
-    base: viteOptions.base,
-    resolve: viteOptions.resolve,
-    // don't copy anything from public folder
-    publicDir: false,
-    build: {
-      sourcemap: viteOptions.build.sourcemap,
-      lib: {
-        entry: options.swSrc,
-        name: 'app',
-        formats: [format],
+  if (format === 'iife') {
+    await build({
+      root: viteOptions.root,
+      base: viteOptions.base,
+      resolve: viteOptions.resolve,
+      mode: options.mode,
+      // don't copy anything from public folder
+      publicDir: false,
+      build: {
+        sourcemap: viteOptions.build.sourcemap,
+        lib: {
+          entry: options.swSrc,
+          name: 'app',
+          formats: [format],
+        },
+        rollupOptions: {
+          ...rollupOptions,
+          plugins,
+          output: {
+            entryFileNames: options.filename,
+          },
+        },
+        outDir: options.outDir,
+        emptyOutDir: false,
       },
-      rollupOptions: {
-        ...rollupOptions,
-        plugins,
-        output: {
-          entryFileNames: options.filename,
+      configFile: false,
+      define,
+    })
+  }
+  else {
+    await build({
+      root: viteOptions.root,
+      base: viteOptions.base,
+      resolve: viteOptions.resolve,
+      mode: options.mode,
```

Comment:
I wonder if we could have a temporary object to put the share options, for future maintenance?

Reply:
sure